### PR TITLE
fix IRTools on Julia 1.6 

### PR DESCRIPTION
**Pull request by simeonschaub**
Still needs https://github.com/JuliaLang/julia/pull/38519. This passes all tests for me locally, but coverage is not great, so there might still be cases I missed.

### CODE REVIEW

File: src/ir/wrap.jl [hunk 1]
```diff
@@ -66,15 +66,30 @@ function IRCode(ir::IR)
...
   for i = 1:length(ci.code)
     ex = ci.code[i]
-    if isexpr(ex, :gotoifnot)
-      push!(bs, ex.args[2])
+    if isgotoifnot(ex)
+      _, dest = destruct_gotoifnot(ex)
+      push!(bs, dest)
       terminator = true
-    elseif isexpr(ex, GotoNode, :return)
+    elseif ex isa GotoNode || isreturn(ex)
```
Comment (oxinabox) [inline src/ir/wrap.jl:92]: Shouldn't this be `&&`?

Comment (simeonschaub) [reply src/ir/wrap.jl:92]: No, I am pretty sure this is correct: https://github.com/FluxML/MacroTools.jl/blob/a03742057f388899833c9000ebf56dbd9616a6d5/src/utils.jl#L39. `ex` can only be either a `GotoNode` or a `ReturnNode`, not both.

**Comment by simeonschaub** [comment]
Nightly tests pass! :tada: Could anyone merge this?

File: src/ir/wrap.jl (hunk 1 above)
Comment (oxinabox) [reply src/ir/wrap.jl:92]: cool

**Comment by femtomc** [comment]
@DhairyaLGandhi

**Comment by DhairyaLGandhi** [comment]
Thanks, this is great! I will run this with Zygote and Flux on 1.6, and get on it

**Comment by simeonschaub** [comment]
FYI, this isn't enough for Zygote yet. I have started working on that in [this branch](https://github.com/simeonschaub/Zygote.jl/tree/sds/fix_on_master) and a corresponding one for ZygoteRules, but I don't currently have time to further look into what it would take to get all Zygote tests to pass on 1.6

**Comment by DhairyaLGandhi** [comment]
Ah, alright, for now let's merge this, thanks again!

**Comment by simeonschaub** [comment]
Would it be possible to also tag a new release?

**Pull request merged**